src/main/java/com/netchill/webservices/api/GenreWs.java

```diff
@@ -29,9 +32,15 @@ public GenreWs(ConfigurationService configurationService, GenreService genreServ
     }
 
     @GET
-    @Path("/id")
-    public Genre getGenreById(@QueryParam("id") Long id) {
-        return this.genreService.getGenreById(id);
+    @Path("/{id}")
+    public Optional<Genre> getGenreById(@PathParam("id") Long id) {
+        Optional<Genre> result = this.genreService.getGenreById(id);
+
+        if (result == null) {
+            throw new WsException(NetchillWsError.RESOURCE_NOT_FOUND);
+        }
```

Comment:
result ne sera jamais null, c'est un optional.

il faut faire :

```suggestion
        if (result.isEmpty()) {
            throw new WsException(NetchillWsError.RESOURCE_NOT_FOUND);
        }
```

Ou 

```
        // Ici on récupère le contenu de l'`Optional`, et s'il est vide on met `null` comme valeur par defaut.
        // Dans d'autre situation tu pourrais mettre autre chose comme valeur par defaut.
        Genre result = this.genreService.getGenreById(id).orElse(null);

        if (result == null) {
            throw new WsException(NetchillWsError.RESOURCE_NOT_FOUND);
        }
```